units/patch_intent.go

```diff
@@ -216,11 +217,33 @@ func (j *patchIntentProcessor) finishPatch(patchDoc *patch.Patch, githubOauthTok
 	return nil
 }
 
-func fetchDiffByURL(URL string) (string, error) {
-	client := util.GetHttpClient()
-	defer util.PutHttpClient(client)
+// buildPatchURL creates a URL to enable downloading patch files through the
+// Github API
+func buildPatchURL(gp *patch.GithubPatch) string {
+	url := &url.URL{
+		Scheme: "https",
+		Host:   "api.github.com",
+		Path: fmt.Sprintf("/repos/%s/%s/pulls/%d.diff", gp.BaseOwner,
+			gp.BaseRepo, gp.PRNumber),
+	}
+
+	return url.String()
+}
+
+func fetchDiffByURL(gh *patch.GithubPatch, token string) (string, error) {
+	client, err := util.GetHttpClientForOauth2(token)
+	if err != nil {
+		return "", nil
+	}
+	defer util.PutHttpClientForOauth2(client)
+
+	req, err := http.NewRequest("GET", buildPatchURL(gh), nil)
+	req.Header.Add("Accept", "application/vnd.github.v3.diff")
+	if err != nil {
+		return "", nil
+	}
 
-	resp, err := client.Get(URL)
+	resp, err := client.Do(req)
 	if err != nil {
```

Comment:
wrap error on line below (is outside the diff, but this seems like a place where we might receive some unexpected response that it would be very useful to wrap)

Reply:
fixed